src/App.vue

```diff
@@ -1,14 +1,104 @@
 <template>
-  <Base/>
+  <v-app light>
+    <Navbar :menu-items="menuItems" />
+    <MobileNavbar :menu-items="menuItems" />
+    <vue-page-transition style="height:100%" >
+    <component :is="layout">
+      <vue-page-transition style="height:100%" >
+        <router-view/>
+      </vue-page-transition>
+    </component>
+    </vue-page-transition>
+  </v-app>
 </template>
 
 <script>
-import Base from './views/Base'
+import Default from '@/components/layouts/Default.vue'
+import Large from '@/components/layouts/Large.vue'
+import None from '@/components/layouts/None.vue'
+import Navbar from '@/components/Navbar.vue'
+import MobileNavbar from '@/components/MobileNavbar.vue'
+
+import { between, required, email } from 'vee-validate/dist/rules'
+import { extend } from 'vee-validate'
+
+extend('passwordMatch', {
+  validate: (value, { other }) => value === other,
+  message: 'Hasła nie są identyczne.',
+  params: [{ name: 'other', isTarget: true }]
+})
+extend('between', {
+  validate: (value, { min, max }) => (value.length > min && value.length < max),
+  message: '{_field_} musi składać się z więcej niż {min} znaków.',
+  params: between.params
+})
+
+extend('require', {
+  validate: required.validate,
+  computesRequired: required.computesRequired,
+  message: 'Pole {_field_} jest wymagane.'
+})
+
+extend('email', {
+  validate: email.validate,
+  message: 'Podany adres email jest niepoprawny.'
+})
 
 export default {
   name: 'App',
   components: {
-    Base
+    Default,
+    Large,
+    None,
+    Navbar,
+    MobileNavbar
+  },
+  data () {
+    return {
+      menuItems: [
+        {
+          title: 'Rejestracja',
+          path: { name: 'RegistrationForm' },
+          icon: 'Face',
+          requireAuth: false
+        },
+        {
+          title: 'Logowanie',
+          path: { name: 'LoginForm' },
+          icon: 'LockOpen',
+          requireAuth: false
+        },
+        {
+          title: 'Wyloguj się',
+          path: { name: 'Home' },
+          icon: 'Logout',
+          requireAuth: true
+        },
+        {
+          title: 'Katalog Quizów',
+          path: { name: 'QuizList', params: { pageNumber: 1 } },
+          icon: 'BookMultiple',
+          requireAuth: false
+        }
+      ]
+    }
+  },
+  computed: {
+    layout () {
+      return (this.$route.meta.layout || process.env.VUE_APP_DEFAULT_LAYOUT)
+    }
   }
 }
 </script>
+
+<style>
```

Comment:
Brakuje chyba słowa kluczowego scoped. Wtedy style zaaplikują Ci się do każdej ikony i loga. Gdzie jest BEM? 
f.e. "navbar__logo"


Jeśli będziesz korzystać z sassa możesz robić takie rzeczy jak:
`.navbar {
  &__logo {
    cursor: pointer;
  }
}`

Także lepsza jest czytelność takiego cssa. Skoro to jest w App to ja nie wiem co to jest za logo i czego.

Reply:
Generalnie wrzuciłem to do App, żeby ostylować logo quizyfy i ikony w navbarze i mobile navbarze. Z logo można by zrobić osobny komponent i tam wrzucić css, natomiast co do ikon to nie mam pomysłu co zrobić, żeby nie kopiować tego samego w tych dwóch komponentach.

Reply:
Wyżej masz odpowiedź już w punkcie 2) lub 3).

Reply:
Uzylem BEM i klas vuetify